__tests__/endpoint.test.js

```diff
@@ -242,3 +242,54 @@ describe("get all /api/users", () => {
       });
   });
 });
+describe("query topic", () => {
+  test("returns array of articles filtered by topic when passed valid topic query  ", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toHaveLength(12);
+        expect(body[0].topic).toBe("mitch");
+      });
+  });
+  test("returns array of articles filtered by topic when passed valid topic query type check  ", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toHaveLength(12);
+        body.forEach((topic) => {
+          expect(typeof topic.author).toBe("string");
+          expect(typeof topic.body).toBe("string");
+          expect(typeof topic.title).toBe("string");
+          expect(typeof topic.article_img_url).toBe("string");
+          expect(typeof topic.body).toBe("string");
+          expect(typeof topic.article_id).toBe("number");
+        });
+      });
+  });
+  test("404 when given a query that does not exist, respond with a not found error", () => {
+    return request(app)
+      .get("/api/articles?topic=socks")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+  test("404 when given a a path with no topics, respond with a not found error", () => {
+    return request(app)
+      .get("/api/comments?topics=")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+  test("404 when given a a path with a comment with no topic, respond with a not found error", () => {
+    return request(app)
+      .get("/api/comments?topics=paper")
```

Comment:
I think this should be `/api/articles?topics=paper` - would this endpoint be an error if the topic exists? Or would we expect our response body to be successful and empty instead?